service/controller/app/v1/resource/clients/resource.go

```diff
@@ -130,6 +126,16 @@ func (r *Resource) addClientsToContext(ctx context.Context, cr v1alpha1.App) err
 		}
 	}
 
+	var tillerNamespace string
+	{
+		// When InCluster is used we use the Tiller deployed in kube-system.
+		if key.InCluster(cr) {
+			tillerNamespace = metav1.NamespaceSystem
+		} else {
+			tillerNamespace = "giantswarm"
+		}
```

Comment:
This is a pain but the in cluster flag is the best way to determine this. Helm 3 helps a bit but we still need to know the namespace for the migration logic.